src/Compilers/CSharp/Portable/Symbols/TupleFieldSymbol.cs

```diff
@@ -0,0 +1,225 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Roslyn.Utilities;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// A TupleFieldSymbol represents the field of a tuple type, such as (int, byte).Item2 or (int a, long b).a
+    /// </summary>
+    internal sealed class TupleFieldSymbol : FieldSymbol
+    {
+        private readonly string _name;
+        private readonly TypeSymbol _type;
+        private readonly TupleTypeSymbol _containingTuple;
+        private readonly int _position;
+        private readonly FieldSymbol _underlyingFieldOpt;
+
+        /// <summary>
+        /// Missing underlying field is handled for error recovery
+        /// A tuple without backing fields is usable for binding purposes, since we know its name and type,
+        /// but caller is supposed to report some kind of error at declaration.
+        ///
+        /// position is 1 for Item1.
+        /// </summary>
+        internal TupleFieldSymbol(string name, TupleTypeSymbol containingTuple, TypeSymbol type, int position, FieldSymbol underlyingFieldOpt)
+        {
+            _name = name;
+            _containingTuple = containingTuple;
+            _underlyingFieldOpt = underlyingFieldOpt;
+            _type = type;
+            _position = position;
+        }
+
+        /// <summary>
+        /// Copy this tuple field, but modify it to use the new containing tuple and underlying type.
+        /// </summary>
+        internal TupleFieldSymbol WithUnderlyingType(TupleTypeSymbol newContainingTuple, NamedTypeSymbol newContainingUnderlyingType)
+        {
+            FieldSymbol newUnderlyingFieldOpt = _underlyingFieldOpt?.OriginalDefinition.AsMember(newContainingUnderlyingType);
+            return new TupleFieldSymbol(_name, newContainingTuple, newUnderlyingFieldOpt.Type, _position, newUnderlyingFieldOpt);
```

Comment:
Appears that `newUnderlyingFieldOpt` can be null here hence `newUnderlyingFieldOpt.Type` is subject to NRE.